src/sage/combinat/permutation.py

```diff
@@ -5431,20 +5433,20 @@ def nth_roots(self, n):
             ...
             ValueError: n must be at least 1
         """
-        
         from sage.combinat.partition import Partitions
         from sage.combinat.set_partition import SetPartitions
         from itertools import product
         from sage.arith.misc import divisors, gcd
 
         def merging_cycles(list_of_cycles):
             """
-            Generate all l-cycles such that its n-th power is the product of cycles in Cycles (which conctains gcd(l, n) cycles of lenght l/gcd(l, n))
+            Generate all l-cycles such that its n-th power is the product
+            of cycles in 'cycles' (which contains gcd(l, n) cycles of length l/gcd(l, n))
             """
             lC = len(list_of_cycles)
             lperm = len(list_of_cycles[0])
             l = lC*lperm
-            perm = [0 for i in range(l)]
+            perm = [0]
```

Comment:
are you sure?  don't you access perm[j*lC] below?

Reply:
oops, indeed, sorry